core/src/main/java/org/apache/iceberg/ManifestFilterManager.java

```diff
@@ -170,16 +172,13 @@ void delete(CharSequence path) {
       return ImmutableList.of();
     }
 
-    // use a common metrics evaluator for all manifests because it is bound to the table schema
-    StrictMetricsEvaluator metricsEvaluator = new StrictMetricsEvaluator(tableSchema, deleteExpression);
```

Comment:
I can no longer use one metrics evaluator so I switched to one per manifest with some caching. I can introduce more caching per spec to avoid creating one expression evaluator per manifest but it will probably complicate things a little. Let me know what everybody thinks.